src/AppInstallerCommonCore/MsixInfo.cpp

```diff
@@ -117,6 +121,68 @@ namespace AppInstaller::Msix
 
             WriteStreamToFile(stream.Get(), size, target, progress);
         }
+
+        // Writes the stream (from current location) to the given file handle.
+        void WriteStreamToFileHandle(IStream* stream, UINT64 expectedSize, HANDLE target, IProgressCallback& progress)
+        {
+            constexpr ULONG bufferSize = 1 << 20;
+            std::unique_ptr<char[]> buffer = std::make_unique<char[]>(bufferSize);
+
+            UINT64 totalBytesRead = 0;
+
+            while (!progress.IsCancelled())
+            {
+                ULONG bytesRead = 0;
+                HRESULT hr = stream->Read(buffer.get(), bufferSize, &bytesRead);
+
+                if (bytesRead)
+                {
+                    // If we got bytes, just accept them and keep going.
+                    LOG_IF_FAILED(hr);
+
+                    THROW_LAST_ERROR_IF(INVALID_SET_FILE_POINTER == SetFilePointer(target, 0, nullptr, FILE_END));
+                    DWORD bytesWritten = 0;
+                    THROW_LAST_ERROR_IF(!WriteFile(target, buffer.get(), bytesRead, &bytesWritten, nullptr));
+                    THROW_HR_IF(E_UNEXPECTED, bytesRead != bytesWritten);
+                    totalBytesRead += bytesRead;
+                    progress.OnProgress(totalBytesRead, expectedSize, ProgressType::Bytes);
+                }
+                else
+                {
+                    // If given a size, and we have read it all, quit
+                    if (expectedSize && totalBytesRead == expectedSize)

```

Comment:
We don't appear to be doing anything to prevent writing more than `expectedSize`.  Is that intentional?